src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/AssemblyBuilderImpl.cs

```diff
@@ -75,24 +77,22 @@ internal void Save(Stream stream)
             }
 
             // Add assembly metadata
-            var metadata = new MetadataBuilder();
-
-            metadata.AddAssembly(
-               metadata.GetOrAddString(value: _assemblyName.Name!),
+            _metadataBuilder.AddAssembly(
+               _metadataBuilder.GetOrAddString(value: _assemblyName.Name!),
                version: _assemblyName.Version ?? new Version(0, 0, 0, 0),
-               culture: _assemblyName.CultureName == null ? default : metadata.GetOrAddString(value: _assemblyName.CultureName),
-               publicKey: _assemblyName.GetPublicKey() is byte[] publicKey ? metadata.GetOrAddBlob(value: publicKey) : default,
+               culture: _assemblyName.CultureName == null ? default : _metadataBuilder.GetOrAddString(value: _assemblyName.CultureName),
+               publicKey: _assemblyName.GetPublicKey() is byte[] publicKey ? _metadataBuilder.GetOrAddBlob(value: publicKey) : default,
                flags: AddContentType((AssemblyFlags)_assemblyName.Flags, _assemblyName.ContentType),
 #pragma warning disable SYSLIB0037 // Type or member is obsolete
                hashAlgorithm: (AssemblyHashAlgorithm)_assemblyName.HashAlgorithm
 #pragma warning restore SYSLIB0037
                );
 
             // Add module's metadata
-            _module.AppendMetadata(metadata);
+            _module.AppendMetadata();
 
             var ilBuilder = new BlobBuilder();
-            WritePEImage(stream, metadata, ilBuilder);
+            WritePEImage(stream, _metadataBuilder, ilBuilder);
```

Comment:
Nit: WritePEImage can be instance method and fetch `_metadataBuilder` from `this`, similar to the rest of methods.